### PR TITLE
fix: golint failures in `pkg/proxy/apis/{config, v1alpha1}` and `pkg/proxy/userspace`

### PR DESCRIPTION
**Pull request by mayankshah1607**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
fix golint failures in `pkg/proxy/apis/{config, v1alpha1}` and `pkg/proxy/userspace`
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/proxy/userspace/roundrobin.go [hunk 1]
```diff
@@ -34,8 +34,10 @@ import (
 )
 
 var (
+	// ErrMissingServiceEntry holds the error message for when the service entry is missing
 	ErrMissingServiceEntry = errors.New("missing service entry")
-	ErrMissingEndpoints    = errors.New("missing endpoints")
+	// ErrMissingEndpoints holds the error message for when the endpoints are missing
+	ErrMissingEndpoints = errors.New("missing endpoints")
```
Comment (aojea) [inline pkg/proxy/userspace/roundrobin.go:40]: should go.fmt warning about this? it seems it has not the same indentation

**Comment by aojea** [comment]
/ok-to-test
please add the release-note NONE to the description

**Comment by aojea** [comment]
you have to squash the commits where you have added and remove the file my mistake, so we only have 1 commit instead of two

File: pkg/proxy/apis/config/types.go [hunk 2]
```diff
@@ -180,11 +180,13 @@ type KubeProxyConfiguration struct {
 // insufficient, this always falls back to the userspace proxy. IPVS mode will be enabled when proxy mode is set to 'ipvs',
 // and the fall back path is firstly iptables and then userspace.
 
+// ProxyMode variable is used to hold the mode in which the proxy will run.
```
Comment (thockin) [inline pkg/proxy/apis/config/types.go:183]: If we're going to fix this comment, shouldn't we fold in the comment just above this?  E.g.

```
ProxyMode defines options for how the proxier implements proxying.  On the Linux platform the available values are ... On the Windows platform the available options are ...
```

The rest of these comments (fallbacks) should go to where this is used.

File: pkg/proxy/userspace/roundrobin.go (hunk 1 above)
Comment (thockin) [reply pkg/proxy/userspace/roundrobin.go:40]: The comment makes it not have to

**Pull request closed without merging**